python/ray/experimental/state/common.py

```diff
@@ -788,6 +818,254 @@ def to_summary(cls, *, tasks: List[Dict]):
+                    DRIVER_TASK_ID_PREFIX
+                ):
+                    summary.append(task_group_by_id[key])
+                else:
+                    parent_task_group = get_or_create_task_group(parent_task_id)
+                    if parent_task_group:
+                        parent_task_group.children.append(task_group_by_id[key])
+
+            return task_group_by_id[key]
+
+        # Step 2: Create the tree structure based on ownership
+        for task in tasks:
+            task_id = task["task_id"]
+
+            task_group = get_or_create_task_group(task_id)
+
+            if not task_group:
+                # We are probably missing data about this task or one of its parents.
+                continue
+
+            state = task["state"]
+            if state not in task_group.state_counts:
+                task_group.state_counts[state] = 0
+            task_group.state_counts[state] += 1
+
+            type_enum = TaskType.DESCRIPTOR.values_by_name[task["type"]].number
+            if type_enum == TaskType.NORMAL_TASK:
+                total_tasks += 1
+            elif type_enum == TaskType.ACTOR_CREATION_TASK:
+                total_actor_scheduled += 1
+            elif type_enum == TaskType.ACTOR_TASK:
+                total_actor_tasks += 1
+
+        def merge_sibings_for_task_group(
+            siblings: List[NestedTaskSummary],
+        ) -> Tuple[NestedTaskSummary, Optional[int]]:
+            """
+            Merges children with the same name into a group if there are more than
+            one child with that name.
+            """
+            if not len(siblings):
+                return siblings, None
+
+            # Group by name
+            groups = {}
+            min_timestamp = None
+
+            for child in siblings:
+                child.children, child_min_timestamp = merge_sibings_for_task_group(
+                    child.children
+                )
+                if child_min_timestamp and child_min_timestamp < (
+                    child.timestamp or sys.maxsize
+                ):
+                    child.timestamp = child_min_timestamp
+
+                if child.name not in groups:
+                    groups[child.name] = NestedTaskSummary(
+                        name=child.name,
+                        key=child.name,
+                        type="GROUP",
+                    )
+                groups[child.name].children.append(child)
+                if child.timestamp and child.timestamp < (
+                    groups[child.name].timestamp or sys.maxsize
+                ):
+                    groups[child.name].timestamp = child.timestamp
+                    if child.timestamp < (min_timestamp or sys.maxsize):
+                        min_timestamp = child.timestamp
+
+            # Return merged siblings
+            return [
+                group if len(group.children) > 1 else group.children[0]
```

Comment:
This line is a bit hard to understand. Can you comment what `group.children[0]` means here?